In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
plt.style.use('ggplot')

In [20]:
data = pd.read_csv('../data/train_fb.csv', index_col='row_id')

In [21]:
data.head()

x       y  accuracy    time    place_id
row_id                                              
0       0.7941  9.0809        54  470702  8523065625
1       5.9567  4.7968        13  186555  1757726713
2       8.3078  7.0407        74  322648  1137537235
3       7.3665  2.5165        65  704587  6567393236
4       4.0961  1.1307        31  472130  7440663949

In [22]:
data.describe()

x             y      accuracy          time      place_id
count  2.911802e+07  2.911802e+07  2.911802e+07  2.911802e+07  2.911802e+07
mean   4.999770e+00  5.001814e+00  8.284912e+01  4.170104e+05  5.493787e+09
std    2.857601e+00  2.887505e+00  1.147518e+02  2.311761e+05  2.611088e+09
min    0.000000e+00  0.000000e+00  1.000000e+00  1.000000e+00  1.000016e+09
25%    2.534700e+00  2.496700e+00  2.700000e+01  2.030570e+05  3.222911e+09
50%    5.009100e+00  4.988300e+00  6.200000e+01  4.339220e+05  5.518573e+09
75%    7.461400e+00  7.510300e+00  7.500000e+01  6.204910e+05  7.764307e+09
max    1.000000e+01  1.000000e+01  1.033000e+03  7.862390e+05  9.999932e+09

In [24]:
# (29118021, 5) for full train dataset
data.shape

(29118021, 5)

In [ ]:
# Looks almost uniform distributes :(
plt.figure(figsize=(12, 12))
plt.scatter(data['x'], data['y'], c=data['place_id'], cmap='Set1', alpha=0.7)
plt.show()

In [29]:
# 108390 categories for full dataset
print('Number of categiries: {0}'.format(len(data['place_id'].unique())))

Number of categiries: 108390


In [31]:
counted_values = data['place_id'].value_counts()

In [37]:
num = 0
for key, val in counted_values.iteritems():
    if val > 1000:
        num += 1
    else:
        break
        
print('Number of categories that occures more than 10 times: {0}').format(num)

Number of categories that occures more than 10 times: 3722


Number of categories that occures more than:
10 times: 107112
100 times: 79569
1000 times: 3722